syntax = "proto3";

service Strelka {
  rpc StreamFile(stream DataRequest) returns (Response) {}

  rpc SendFile(DataRequest) returns (Response) {}

  rpc SendLocation(LocationRequest) returns (Response) {}
}

message DataRequest {
  string uid = 1;
  bytes data = 2;
  string filename = 3;
  string source = 4;
  repeated string flavors = 5;
  map<string, string> metadata = 6;
  bool log_scan = 7;
}

message LocationRequest {
  string uid = 1;
  map<string, string> location = 2;
  string filename = 3;
  string source = 4;
  repeated string flavors = 5;
  map<string, string> metadata = 6;
  bool log_scan = 7;
}

message Response {
  string uid = 1;
  float elapsed = 2;
  string result = 3;
}
